components/raftstore/src/store/peer.rs

```diff
@@ -6204,7 +6271,54 @@ pub fn make_transfer_leader_response() -> RaftCmdResponse {
 
 // The Raft message context for a MsgTransferLeader if it is a reply of a
 // TransferLeader command.
-pub const TRANSFER_LEADER_COMMAND_REPLY_CTX: &[u8] = &[1];
+const TRANSFER_LEADER_COMMAND_REPLY_CTX: &[u8] = &[1];
+
+/// The Raft message context for a MsgTransferLeader
+#[derive(Debug, PartialEq)]
+pub enum TransferLeaderContext {
+    /// None is a zero-length context.
+    None,
+    /// A reply of a AdminCmd TransferLeader.
+    /// Tag: 1.
+    CommandReply,
+}
+
+impl TransferLeaderContext {
+    const TAG_COMMAND_REPLY: u8 = 1;
+    pub fn from_bytes(mut ctx: &[u8]) -> Result<TransferLeaderContext> {
+        if ctx.is_empty() {
+            return Ok(TransferLeaderContext::None);
+        }
+        match box_try!(ctx.read_u8()) {
+            Self::TAG_COMMAND_REPLY => Ok(TransferLeaderContext::CommandReply),
+            tag => Err(box_err!("invalid tag: {}", tag)),
+        }
+    }
+
+    pub fn to_bytes(&self) -> Result<Bytes> {
+        match self {
+            TransferLeaderContext::None => Ok(Bytes::new()),
+            TransferLeaderContext::CommandReply => {
+                Ok(Bytes::from_static(TRANSFER_LEADER_COMMAND_REPLY_CTX))
+            }
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct TransferLeaderState {
+    /// lead_transferee if leader is in a leadership transferring.
+    /// Only leader can update this field, and it is meaningful only when a peer
+    /// is a leader or a leader is stepping down.
+    pub lead_transferee: u64,
```

Comment:
Should it be `leader_transferee` ?
```suggestion
    pub leader_transferee: u64,
```